File: src/dataset/javascript/difference-of-squares_exercism-javascript-programmiri/claude/ClaudeSonnet4_difference-of-squares_v3.js
export class Squares {
  constructor(number) {
    this.number = number;
    this._sumOfSquares = null;
    this._squareOfSum = null;
  }

  get sumOfSquares() {
    if (this._sumOfSquares === null) {
      this._sumOfSquares = (this.number * (this.number + 1) * (2 * this.number + 1)) / 6;
    }
    return this._sumOfSquares;
  }

  get squareOfSum() {
    if (this._squareOfSum === null) {
      const sum = (this.number * (this.number + 1)) / 2;
      this._squareOfSum = sum * sum;
    }
    return this._squareOfSum;
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares;
  }
}

File: src/dataset/c/difference-of-squares_exercism-maxclaus/claude/ClaudeSonnet4_difference_of_squares_v3.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int number) {
  return (number * (number + 1) * ((number << 1) + 1)) / 6;
}

unsigned int square_of_sum(unsigned int number) {
  unsigned int sum = (number * (number + 1)) >> 1;
  return sum * sum;
}

unsigned int difference_of_squares(unsigned int number) {
  unsigned int sum = (number * (number + 1)) >> 1;
  return sum * sum - (number * (number + 1) * ((number << 1) + 1)) / 6;
}

unsigned int sum_of_squares_v1(unsigned int number) {
  unsigned int sum = 0;
  for (unsigned int i = 1; i <= number; i++)
    sum += (i * i);
  return sum;
}

unsigned int square_of_sum_v1(unsigned int number) {
  unsigned int sum = 0;
  for (unsigned int i = 1; i <= number; i++)
    sum += i;
  return sum * sum;
}

unsigned int difference_of_squares_v1(unsigned int number) {
  unsigned int result_sum_of_squares = 0;
  unsigned int result_square_of_sum = 0;
  for (unsigned int i = 1; i <= number; i++) {
    result_sum_of_squares += (i * i);
    result_square_of_sum += i;
  }
  return (result_square_of_sum * result_square_of_sum) - result_sum_of_squares;
}

unsigned int sum_of_squares_v2(unsigned int number) {
  return (number * (number + 1) * ((number << 1) + 1)) / 6;
}

unsigned int square_of_sum_v2(unsigned int number) {
  unsigned int sum = (number * (number + 1)) >> 1;
  return sum * sum;
}

unsigned int difference_of_squares_v2(unsigned int number) {
  unsigned int sum = (number * (number + 1)) >> 1;
  return sum * sum - (number * (number + 1) * ((number << 1) + 1)) / 6;
}